src/libraries/System.Private.CoreLib/src/System/Globalization/NumberFormatInfo.cs

```diff
@@ -155,10 +161,23 @@ private static void VerifyDigitSubstitution(DigitShapes digitSub, string propert
         }
 
         internal bool HasInvariantNumberSigns => _hasInvariantNumberSigns;
+        internal bool AllowHyphenDuringParsing => _allowHyphenDuringParsing;
 
-        private void UpdateHasInvariantNumberSigns()
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void InitializeInvariantAndNegativeSignFlags()
         {
             _hasInvariantNumberSigns = _positiveSign == "+" && _negativeSign == "-";
+            _allowHyphenDuringParsing = _negativeSign.Length == 1 &&
+                                        _negativeSign[0] switch {
```

Comment:
Nit: add code comment explaining where this test came from. If we wanted to, we could even have a unit test that cycles through this entire list and ensures a hyphen matches every one of them. We have some unit test helper libraries that parse the Unicode data files, used primarily for Rune and StringInfo and S.T.E.W, but if there's interest we could also adapt it for this.

(Whether we should have a unit test that iterates through this really depends on whether this list is populated from one of the Unicode data files.)